WASM support broken due to incompatible dependencies

Following instructions at https://jolt.a16zcrypto.com/usage/wasm_support.html, I was unable to compile my project to wasm.

**Steps to reproduce**

1. Make a new project (`jolt new foo --wasm`)
2. Ensure `fib(n: 32) -> u128` in `guest/src/lib.rs` has `#[jolt::provable(wasm)]`
3. Build the project `jolt build-wasm`

**Expected result**
I expect a wasm bin to be compiled and output to `target/wasm32-unknown-unknown/release` as well as an `index.html` and a `pkg` dir with some wasm-pack stuff.

**Actual result**
`index.html` is created, but wasm compilation fails as it encounters some `nix` modules within the `tracing-profile` dependency:

```
error[E0433]: failed to resolve: could not find `sys` in `nix`
 --> ~/.cargo/git/checkouts/tracing-profile-ea11638690e1685c/6d986ca/src/data/span_metadata.rs:4:10
  |
4 | use nix::sys::time::TimeSpec;
  |          ^^^ could not find `sys` in `nix`

error[E0433]: failed to resolve: could not find `sys` in `nix`
 --> ~/.cargo/git/checkouts/tracing-profile-ea11638690e1685c/6d986ca/src/layers/csv.rs:1:10
  |
1 | use nix::sys::time::TimeValLike;
  |          ^^^ could not find `sys` in `nix`

error[E0432]: unresolved import `nix::time`
 --> ~/.cargo/git/checkouts/tracing-profile-ea11638690e1685c/6d986ca/src/layers/csv.rs:2:10
  |
2 | use nix::time::{clock_gettime, ClockId};
  |          ^^^^ could not find `time` in `nix`
```

`cargo tree -i tracing profile` reveals the tree:

```
cargo tree -i tracing-profile              
tracing-profile v0.8.0 (https://gitlab.com/IrreducibleOSS/tracing-profile.git?branch=main#6d986ca1)
└── binius_utils v0.1.0 (https://gitlab.com/UlvetannaOSS/binius#a0a71393)
    └── binius_field v0.1.0 (https://gitlab.com/UlvetannaOSS/binius#a0a71393)
        └── jolt-core v0.1.0 (https://github.com/a16z/jolt#e08998e8)
            └── jolt-sdk v0.1.0 (https://github.com/a16z/jolt#e08998e8)
                ├── guest v0.1.0 (~/foo/guest)
                │   └── foo v0.1.0 (~/foo)
                └── foo v0.1.0 (~/foo)
```

--- a/jolt-core/Cargo.toml
+++ b/jolt-core/Cargo.toml
@@ -27,7 +27,6 @@ ark-serialize = { version = "0.4.2", default-features = false, features = [
     "derive",
 ] }
 ark-std = { version = "0.4.0" }
-binius-field = { git = "https://gitlab.com/UlvetannaOSS/binius", package = "binius_field" }
 clap = { version = "4.3.10", features = ["derive"] }
 enum_dispatch = "0.3.12"
 fixedbitset = "0.5.0"
@@ -66,7 +65,7 @@ indicatif = "0.17.8"
 common = { path = "../common" }
 tracer = { path = "../tracer" }
 bincode = "1.3.3"
-bytemuck = "1.15.0"
+bytemuck = "1.19.0"
 hex = "0.4.3"
 tokio = { version = "1.38.0", optional = true }
 alloy-primitives = "0.7.6"
@@ -111,6 +110,7 @@ host = ["dep:reqwest", "dep:tokio"]
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
 memory-stats = "1.0.0"
 tokio = { version = "1.38.0", optional = true, features = ["rt-multi-thread"] }
+binius-field = { git = "https://gitlab.com/UlvetannaOSS/binius", package = "binius_field" }
 
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]

--- a/jolt-core/src/field/mod.rs
+++ b/jolt-core/src/field/mod.rs
@@ -91,4 +91,5 @@ where
 }
 
 pub mod ark;
+#[cfg(not(target_arch = "wasm32"))]
 pub mod binius;

--- a/src/build_wasm.rs
+++ b/src/build_wasm.rs
@@ -53,8 +53,9 @@ fn preprocess_and_save(func_name: &str, attributes: &Attributes, is_std: bool) -
 }
 
 fn extract_provable_functions() -> Vec<FunctionAttributes> {
-    let content = fs::read_to_string("guest/src/lib.rs").expect("Unable to read file");
-    let syntax: syn::File = syn::parse_file(&content).expect("Unable to parse file");
+    let guest_path = Path::new("guest/src/lib.rs");
+    let content = fs::read_to_string(guest_path).expect(&format!("Unable to read file: {:?}", guest_path));
+    let syntax: syn::File = syn::parse_file(&content).expect(&format!("Unable to parse file: {:?}", guest_path));
 
     syntax
         .items

--- a/src/build_wasm.rs
+++ b/src/build_wasm.rs
@@ -54,8 +54,10 @@ fn preprocess_and_save(func_name: &str, attributes: &Attributes, is_std: bool) -
 
 fn extract_provable_functions() -> Vec<FunctionAttributes> {
     let guest_path = Path::new("guest/src/lib.rs");
-    let content = fs::read_to_string(guest_path).expect(&format!("Unable to read file: {:?}", guest_path));
-    let syntax: syn::File = syn::parse_file(&content).expect(&format!("Unable to parse file: {:?}", guest_path));
+    let content = fs::read_to_string(guest_path)
+        .unwrap_or_else(|_| panic!("Unable to read file: {:?}", guest_path));
+    let syntax: syn::File = syn::parse_file(&content)
+        .unwrap_or_else(|_| panic!("Unable to parse file: {:?}", guest_path));
 
     syntax
         .items

